src/generators/java/renderers/ClassRenderer.ts

```diff
@@ -82,14 +82,23 @@ export const JAVA_DEFAULT_CLASS_PRESET: ClassPresetType<JavaOptions> = {
   self({ renderer }) {
     return renderer.defaultSelf();
   },
-  property({ property }) {
+  property({ renderer, property }) {
+    const constValue = renderer.getConstValue(property);
+
+    if (constValue) {
+      return `private final ${property.property.type} ${property.propertyName} = ${property.property.type}.${constValue.key};`;
+    }
+
     return `private ${property.property.type} ${property.propertyName};`;
   },
   getter({ property }) {
     const getterName = `get${FormatHelpers.toPascalCase(property.propertyName)}`;
     return `public ${property.property.type} ${getterName}() { return this.${property.propertyName}; }`;
   },
-  setter({ property }) {
+  setter({ renderer, property }) {
+    if (renderer.getConstValue(property)) {
```

Comment:
The better structure would have been `property.hasConstantValue()` or something similar, but I dont see a way around it, can you? 🤔